pkg/agent/controller/networkpolicy/priority.go

```diff
@@ -0,0 +1,157 @@
+func (pa *priorityAssigner) getPriorityZoneIndex(p types.Priority) int32 {
+	floorPriority := int32(math.Floor(p.PolicyPriority))
+	if floorPriority > pa.numPriorityZones-1 {
+		floorPriority = pa.numPriorityZones - 1
+	}
+	return floorPriority
+}
+
+// getPriorityZoneStart returns the starting OF priority for the priorityZone for the input.
+func (pa *priorityAssigner) getPriorityZoneStart(p types.Priority) uint16 {
+	priorityIndex := pa.getPriorityZoneIndex(p)
+	return DefaultTierStart - pa.priorityOffset*uint16(priorityIndex)
+}
+
+// getPriorityZoneSize returns the size of the priorityZone for the input.
+func (pa *priorityAssigner) getPriorityZoneSize(p types.Priority) uint16 {
+	zoneStart := pa.getPriorityZoneStart(p)
+	if zoneStart-pa.priorityOffset < PriorityBottomCNP {
+		return zoneStart - PriorityBottomCNP
+	}
+	return pa.priorityOffset
+}
+
+// sortPriorities sorts a list of priorities.
+func (pa *priorityAssigner) sortPriorities(priorities []types.Priority) {
+	sort.Slice(priorities, func(i, j int) bool {
+		if priorities[i].PolicyPriority == priorities[j].PolicyPriority {
+			return priorities[i].RulePriority < priorities[j].RulePriority
+		}
+		return priorities[i].PolicyPriority < priorities[j].PolicyPriority
+	})
+}
+
+// getIndexSamePriorityZone returns a list of sorted priorities that needs to be present in the same
+// priority zone of the input priority.
+func (pa *priorityAssigner) getIndexSamePriorityZone(p types.Priority) []types.Priority {
+	affected := []types.Priority{p}
+	for k := range pa.priorityMap {
+		if pa.getPriorityZoneStart(k) == pa.getPriorityZoneStart(p) {
+			affected = append(affected, k)
+		}
+	}
+	pa.sortPriorities(affected)
+	return affected
+}
+
+// syncPriorityZone computes the new expected OF priorties for each priority in the same priority zone
+// of the input priority, and returns installed priorities that need to be re-assigned if necessary.
+func (pa *priorityAssigner) syncPriorityZone(p types.Priority) (*uint16, map[uint16]uint16, error) {
+
+	// newPriority is the OF priority to be assigned for a new priority.
+	// For priority Release, newPriority returned should be nil.
+	var newPriority uint16
+	// priorityUpdates stores all the OF priority re-assignments to be performed by client
+	priorityUpdates := map[uint16]uint16{}
+
+	affected := pa.getIndexSamePriorityZone(p)
+	if uint16(len(affected)) > pa.getPriorityZoneSize(p) {
+		// TODO: Dynamically adjust priorityZone size to handle overflow
+		return nil, priorityUpdates, fmt.Errorf("priorityZone for [%v %v) has overflowed",
+			pa.getPriorityZoneIndex(p), pa.getPriorityZoneIndex(p)+1)
+	}
+	for offset, priority := range affected {
+		computedPriority := pa.getPriorityZoneStart(p) - uint16(offset)
+		oldOFPriority, updateExisting := pa.priorityMap[priority]
+		if updateExisting && computedPriority != oldOFPriority {
+			klog.V(2).Infof("Original priority %d needs to be reassigned %d now.", oldOFPriority, computedPriority)
```

Comment:
nit: Remove ending period here and in other logs

Reply:
Will address this is the CNP E2E patch.